### PR TITLE
Instance Composer Test part 2

### LINKED ISSUE #5093: composer tests

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

Part Two of Instance Composer Tests + some minor improvements that I noticed along the way

closes #5093 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/UI/Components/Diagram/Canvas.test.tsx [hunk 1]
```diff
@@ -526,4 +696,57 @@ describe("Canvas.tsx", () => {
...
+  it("creates valid object for order_api when creating entity from scratch", () => {});
+  it("creates valid object for order_api when editing entity", () => {});
+
+  // it("zoom-in/zoom-out", async () => {
+  //   // TODO: Resolve "TypeError: viewport.getCTM is not a function"
+  //   const component = setup();
+  //   render(component);
+
+  //   const paper = document.querySelector(".joint-layers") as Element;
+  //   //default value, that is set in the browser
```
Comment (LauraStordeur) [inline src/UI/Components/Diagram/Canvas.test.tsx:736]: Is it the same for all browsers?

File: src/UI/Utils/useFeatures.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ const extractFeatures = (fileContent) => {
 };
 
 const useFeatures = (): string[] => {
-  const [features, setFeatures] = useState([]);
+  const [features, setFeatures] = useState(["instanceComposer"]);
```
Comment (LauraStordeur) [inline src/UI/Utils/useFeatures.tsx:14]: I don't think you meant to commit this ;)

File: src/UI/Components/Diagram/Canvas.test.tsx (hunk 1 above)
Comment (matborowczyk) [reply src/UI/Components/Diagram/Canvas.test.tsx:736]: I think i should rename that for "set by library" as it indeed is the same across webpages but thought like why it should be set by browser in the first place :D

File: src/UI/Components/Diagram/Canvas.test.tsx [hunk 3]
```diff
@@ -408,6 +516,68 @@ describe("Canvas.tsx", () => {
...
+
+  //   await act(async () => {
+  //     await user.click(shapes[0]);
+  //   });
+
+  //   const handle = document.querySelector('[data-action="link"]') as Element;
+  //   await act(async () => {
+  //     await user.pointer(
+  //       // click at link element
+  //       { keys: "[MouseLeft>]", target: handle },
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Components/Diagram/Canvas.test.tsx:569]: this pointerName step was throwing error for me which I could not resolve, that's why I moved it into the queue.

File: src/UI/Components/Diagram/Canvas.test.tsx [hunk 4]
```diff
@@ -302,6 +341,75 @@ describe("Canvas.tsx", () => {
...
+    const attrs = await screen.findByJointSelector("labelsGroup_1");
+
+    expect(attrs.childNodes).toHaveLength(4);
+
+    const button = await screen.findByJointSelector("button");
+    await act(async () => {
+      await user.click(button);
+    });
+
+    const attrsTwo = await screen.findByJointSelector("labelsGroup_1");
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Components/Diagram/Canvas.test.tsx:358]: unfortunately that doesn't work there

File: src/UI/Components/Diagram/Canvas.test.tsx [hunk 5]
```diff
@@ -526,4 +696,57 @@ describe("Canvas.tsx", () => {
...
+  //   const component = setup(mockedInstanceWithReferences);
+  //   render(component);
+
+  //   await screen.findAllByJointSelector("info");
+  //   const connectors = document.querySelectorAll('[data-type="app.Link"]');
+
+  //   expect(connectors).toHaveLength(3);
+
+  //   fireEvent.mouseEnter(connectors[0]);
+  //   //Delay has to be done as library base itself on listeners that are async
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Components/Diagram/Canvas.test.tsx:711]: I found resolution that won't include spying on these events as there are multiple of them, and propose `await new Promise(process.nextTick);`

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**